controllers/version_update.go

```diff
@@ -45,23 +56,12 @@ func handleVersionUpdate(master *v1alpha1.CDAPMaster, labels map[string]string,
 	}
 
 	// Update backend service image version
-	curVersion, err := getCurrentImageVersion(master)
-	if err != nil {
-		return nil, err
-	}
-	newVersion, err := getNewImageVersion(master)
-	if err != nil {
-		return nil, err
-	}
 	if len(curVersion.rawString) == 0 {
 		setImageToUse(master)
 		return []reconciler.Object{}, nil
 	}
 
-	switch compareVersion(curVersion, newVersion) {
-	case -1:
-		// Upgrade case
-
+	if versionComparison < 0 { // Upgrade case
```

Comment:
nit: please add comments above the code instead of inline. Comment sentences should end with a period (`.`)

Fix t/o